src/plone/restapi/services/vocabularies/get.py

```diff
@@ -86,7 +86,8 @@ def reply(self):
                 404, "Not Found", f"The vocabulary '{vocabulary_name}' does not exist"
             )
 
-        vocabulary = factory(self.context)
+        query = self.request.form.get("title", "")
+        vocabulary = factory(self.context, query=query)
```

Comment:
@tiberiuichim This won't work with many vocabularies, which is why a lot of tests broke.

The IVocabularyFactory interface specifies that a vocab factory is called with a context, but not other parameters: https://github.com/zopefoundation/zope.schema/blob/master/src/zope/schema/interfaces.py#L1087

The addition of the `query` parameter seems to be something that was made up for the vocabulary factories in plone.app.vocabularies.principals without following any formal interface. This makes it hard to detect here how the factory is supposed to be called.

To really sort this out, we need to define a new interface like IQueryableVocabularyFactory. Then plone.app.vocabularies can define that these vocabulary factories implement that interface, and plone.restapi can be updated to pass the query to factories which provide that interface.